packages/cli/src/commands/env/pull.ts

```diff
@@ -136,11 +138,23 @@ export default async function pull(
     output.log('No changes found.');
   }
 
+  let isGitIgnoreUpdated = false;
+  if (filename === '.env.local') {
+    // When the file is `.env.local`, we also add it to `.gitignore`
+    // to avoid accidentally committing it to git.
+    // We use '.env*.local' to match the default .gitignore from
+    // create-next-app template. See:
+    // https://github.com/vercel/next.js/blob/06abd634899095b6cc28e6e8315b1e8b9c8df939/packages/create-next-app/templates/app/js/gitignore#L28
+    const repoLink = await getRepoLink(cwd);
```

Comment:
Let's pass the `link` in from `env/index.ts` as a parameter instead?

Reply:
@TooTallNate The `link` in `env/index.ts` is not the same. Its a `ProjectLinkResult` type.

https://github.com/vercel/vercel/blob/ef30a46c037adfdf31cdd1e10642130ad41ae8ea/packages/cli/src/commands/env/index.ts#L147

However we need the `RepoLink` type.

Reply:
It's narrowed to `ProjectLinked` by the time `pull()` is called, which contains the `repoRoot` prop.

Reply:
Oh nice I didn't know that `repoRoot` was there too 👍 

Updated in [fee8c02](https://github.com/vercel/vercel/pull/10085/commits/fee8c027a7be1094ec47a2f72d1bb19286598370)